impl/src/main/java/io/kroxylicious/testing/kafka/common/Utils.java

```diff
@@ -53,28 +58,54 @@ public static Stream<ServerSocket> preAllocateListeningSockets(int num) {
     }
 
     public static void awaitExpectedBrokerCountInCluster(Map<String, Object> connectionConfig, int timeout, TimeUnit timeUnit, Integer expectedBrokerCount) {
-        try (Admin admin = Admin.create(connectionConfig)) {
-            Awaitility.await()
-                    .pollDelay(Duration.ZERO)
-                    .pollInterval(1, TimeUnit.SECONDS)
-                    .atMost(timeout, timeUnit)
-                    .ignoreExceptions()
-                    .until(() -> {
-                        log.info("describing cluster: {}", connectionConfig.get("bootstrap.servers"));
-                        final Collection<Node> nodes;
-                        try {
-                            nodes = admin.describeCluster().nodes().get(10, TimeUnit.SECONDS);
-                            log.info("got nodes: {}", nodes);
-                            return nodes;
-                        }
-                        catch (InterruptedException | ExecutionException e) {
-                            log.warn("caught: {}", e.getMessage(), e);
-                        }
-                        catch (TimeoutException te) {
-                            log.warn("Kafka timed out describing the the cluster");
-                        }
-                        return Collections.emptyList();
-                    }, Matchers.hasSize(expectedBrokerCount));
+        var knownReady = Collections.synchronizedSet(new HashSet<String>());
+        var toProbe = Collections.synchronizedSet(new HashSet<String>());
+
+        var originalBootstrap = String.valueOf(connectionConfig.get(BOOTSTRAP_SERVERS_CONFIG));
+        toProbe.addAll(Arrays.asList(originalBootstrap.split(",")));
+
+        while(knownReady.size() < expectedBrokerCount) {
+            if (toProbe.isEmpty()) {
+                throw new IllegalArgumentException(String.format("Ran out of addresses to probe before cluster ready: %s", originalBootstrap));
+            }
+            var probeAddress = toProbe.iterator().next();
+
+            var copy = new HashMap<>(connectionConfig);
+            copy.put(BOOTSTRAP_SERVERS_CONFIG, probeAddress);
+
+            try (Admin admin = Admin.create(copy)) {
+                Awaitility.await()
+                        .pollDelay(Duration.ZERO)
+                        .pollInterval(1, TimeUnit.SECONDS)
```

Comment:
Seeing this again makes me think we should configure the awaitility defaults rather than customising the specific call, but that's not really part of this PR's scope.